src/rgw/rgw_sal_sfs.cc

```diff
@@ -508,6 +510,78 @@ void SFStore::filesystem_stats_updater_main(
   }
 }
 
+const std::vector<std::function<MetricsStatusPage::ScalarMetricFunction>>&
+SFStore::custom_metric_fns() {
+  sqlite3* sqlite_db = db_conn->first_sqlite_conn;
+  auto make_sqlite3_status_fn = [&](const std::string& name, int op,
+                                    bool use_highwater = false) {
+    return [sqlite_db, op, use_highwater, name]() {
+      int current;
+      int highwater;
+      int ret = SQLITE_ERROR;
+      if (sqlite_db) {
+        ret = sqlite3_db_status(sqlite_db, op, &current, &highwater, false);
+      }
+      if (ret != SQLITE_OK) {
+        return std::make_tuple(
+            perfcounter_type_d::PERFCOUNTER_U64, name, std::nan("error")
+        );
+      }
+      return std::make_tuple(
+          perfcounter_type_d::PERFCOUNTER_U64, name,
+          static_cast<double>(use_highwater ? highwater : current)
+      );
+    };
+  };
+
+  static std::vector<std::function<MetricsStatusPage::ScalarMetricFunction>>
```

Comment:
Question:
why returning a const reference? (and having to declare this as static)
The compiler should perform a move when returning the `std::vector` or even elude the move itself, right?

Reply:
This started with a different ownership story, but since the status pages actually copy the std functions to themselves this really should not be the static / const ref stunt. Will refactor